mod/paas_integration/start.php

```diff
@@ -127,4 +127,55 @@ function address_validation($address) {
     }
 
     return null;
+  }
+
+  function waiting_on_approval($guid){
+
+    $dbprefix = elgg_get_config("dbprefix");
+    $service_url = elgg_get_plugin_setting("graphql_client", "paas_integration");
+
+    $result = get_data_row("SELECT pleio_guid FROM {$dbprefix}users_entity WHERE guid = $guid");
+    if ($result->pleio_guid) {
```

Comment:
While I can't think of any way this would be false without data corruption or something like that (connection / db errors should error out before reaching here) maybe this would work better as something like

```php
    if ($result->pleio_guid) {
       // error handling
       return
    }
     $gcID = $result->pleio_guid;
```
to cover that case earlier than when it's sent to paas as an empty / uninitialized variable, assuming I didn't miss something.